google/cloud/aiplatform/metadata/execution.py

```diff
@@ -57,3 +60,23 @@ def _update_resource(
         cls, client: utils.MetadataClientWithOverride, resource: proto.Message,
     ) -> proto.Message:
         return client.update_execution(execution=resource)
+
+    def add_artifact(
+        self, artifact: str, input: bool,
```

Comment:
artifact_name maybe to be consistent with other methods?

Reply:
Updated artifact to artifact_name.

For type, I think it is better to keep it required as there is no real default behavior here. We would likely use this to assign artifact as an Input of a execution later, making it default as output could cause unexpected issues.